### PR TITLE
Remove service principal authentication from McrStatusClient

### PR DESCRIPTION
**Pull request by lbussell**
Part of #1264

I haven't tested this in-place yet since it only runs on real builds, so marking as a draft for now. Early next week I'll run a test by modifying a pipeline to skip publishing and just check the status of some existing build.

### CODE REVIEW

**Comment by lbussell** [comment]
Tried to queue a test but I'm not sure it did anything [internal link] - https://dev.azure.com/dnceng/internal/_build/results?buildId=2444600&view=logs&j=739bb7ec-7296-5b40-7dff-ee3297bfdaa4&t=487ef77f-c8ee-5b2f-147b-5cff06a12f4e

Any other tips for testing this @mthalman? Or since it wouldn't block publishing should we just merge it and revert if there are any big issues?

**Comment by mthalman** [comment]
> Tried to queue a test but I'm not sure it did anything [internal link] - https://dev.azure.com/dnceng/internal/_build/results?buildId=2444600&view=logs&j=739bb7ec-7296-5b40-7dff-ee3297bfdaa4&t=487ef77f-c8ee-5b2f-147b-5cff06a12f4e
> 
> Any other tips for testing this @mthalman? Or since it wouldn't block publishing should we just merge it and revert if there are any big issues?

It didn't do anything because no images were published (due to image caching). You could test this by running Image Builder locally targeting the image info file from a previous publish run.

File: src/Microsoft.DotNet.ImageBuilder/src/AuthHelper.cs [hunk 1]
```diff
@@ -16,5 +18,12 @@ public static async Task<string> GetAadAccessTokenAsync(string resource, string
                 resource, new ClientCredential(username, password));
             return result.AccessToken;
         }
+
+        public static async Task<string> GetMsiAadAccessTokenAsync(string resource)
+        {
+            ManagedIdentityCredential credential  = new ManagedIdentityCredential();
```
Comment (mthalman) [inline src/Microsoft.DotNet.ImageBuilder/src/AuthHelper.cs:24]: Consider changing this to `DefaultAzureCredential`, if that works. That will allow testing of this command to still work on a dev machine.

Comment (lbussell) [reply src/Microsoft.DotNet.ImageBuilder/src/AuthHelper.cs:24]: It's working with default. I'll update it.

**Review by mthalman — APPROVED** [review]

**Pull request merged**